src/test/javascript/spec/component/plagiarism/plagiarism-inspector.component.spec.ts

```diff
@@ -37,12 +36,12 @@ jest.mock('app/shared/util/download.util', () => ({
 }));
 
 const generateCsv = jest.fn();
-
-jest.mock('export-to-csv', () => ({
-    ExportToCsv: jest.fn().mockImplementation(() => ({
-        generateCsv,
-    })),
-}));
+jest.mock('export-to-csv', () => {
+    class MockExportToCsv {
+        generateCsv = generateCsv;
+    }
+    return { ExportToCsv: MockExportToCsv };
+});
```

Comment:
We define this quite often. Shouldn't we extract the definition of the mock and mock the function then directly on the created class as usual?

Reply:
AFAIK there isn't a way to mock external dependencies without having `jest.mock` in the test file itself. The only thing we could extract is the `MockExportToCSV` class, which doesn't seem to be worth it I think